### ISSUE TITLE
Broken GitHub Integration?

### ISSUE THREAD

**Issue by nurse-the-code** [GitbookIO/gitbook#2335]
## Bug description
On the hosted version of GitBook (available at https://app.gitbook.com/), I seemingly can't sync
 with GitBook.
_Please describe._  
_If this affects the front-end, screenshots would be of great help._

## How to reproduce

1. Have a shared workspace using the free hosting tier (and be a member of an organization)
2. Make sure your account is authenticated with Google (first) AND GitHub (second)
3. Follow the instructions in this video: https://www.youtube.com/watch?v=Fm5hYBsRSXo
4. Click on either the "Sync with GitHub / GitLab" button available on the space or the "Synchronize with Git" button in the right hand drop down menu.
5. Either way you should find yourself on the integrations tab, with no clear way to synchronize with GitHub and the UI will be missing the GitHub/GitLab integration module that was in the YouTube video.

**Comment by SugeethJSA** [comment]
I am also facing the same issue. In my situation, the GitHub integration would authenticate and do the initial sync. However, after I close the GitBook tab and open the integrations page, it shows "errors" in the repository linking dropdown box.

Later, when I edit something in GitBook, it would sync with GitHub. However, if I edit something in GitHub, it will not sync back with GitBook until manually start the sync again. Sometimes it would only sync my commits, but if anyone else had made a commit to the documents section, then it would not sync back to GitBook at all.

It would be helpful to know how does the GitHub integration actually work, like if I commit something on GitHub, would the GitBook bot immediately pick up that commit and sync it back to the GitBook end; or would it periodically scan for new commits and then sync? The reason is that I'd like my sync to happen immediately when someone makes a commit. My group and I deal with time-sensitive academic matters that get updated periodically, and the delay in sync would and could cause someone to do the wrong thing, even if the mistake has been corrected or a new edit has been pushed. I wouldn't want any person to believe that they have been misled. I hope you do understand.

And one more thing, GitBook doesn't need to import the entire repository back into GitBook when syncing with GitHub. I guess it should just scan for new commits and then import only that commit back into GitBook. The reason why I'm suggesting this is that my sync (when it works) keeps getting slower and slower every time the repository gets larger and larger.

The documentation on the GitBook public site does not explain how the sync works. It just mentions the steps to set it up and some general details. It would easier for us to diagnose or see if we had made any mistake to configure this GitBook-GitHub bidirectional sync. I would also like to know if there are any rate limits for the bi-directional sync.

Hope you do understand.

**Comment by addisonschultz** [comment]
Hi @SugeethJSA - thanks for the detailed description. To answer your question, any Pull Requests or commits made into the `main` branch of your repository will be synced with GitBook. It should automatically sync once the commit has been made to `main`. 

This works by the GitBook app existing within the organization of the repository being used. 

Do you have this app installed in your organization? https://github.com/marketplace/gitbook-com

**Comment by SugeethJSA** [comment]
Yeah, did all that. But I guess the sync is still not working.

**Comment by addisonschultz** [comment]
If you can reach out to support (@) gitbook.com we can help you look into it for your specific case!

**Comment by SugeethJSA** [comment]
Yeah, did it, @addisonschultz . Turns out that the bidirectional sync wasn't working due to a cache of an older sync that hadn't been cleared yet. Thankfully, this cache was cleared by the support team.
I think it would nice if we had access to clear our own cache for our space to fix sync if needed. Correct me if I'm wrong, but this kinda feels like the set of basic troubleshooting options that creators should have access to, in order to reduce support requests. It would reduce some load for the customer support team.
